src/app/(proper_react)/MobileShell.tsx

```diff
@@ -0,0 +1,92 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+"use client";
+
+import { ReactNode, useState } from "react";
+import Link from "next/link";
+import Image from "next/image";
+import { Session } from "next-auth";
+import styles from "./MobileShell.module.scss";
+import monitorLogo from "./images/monitor-logo.webp";
+import { CloseBigIcon, ListIcon } from "../components/server/Icons";
+import { useL10n } from "../hooks/l10n";
+import { PageLink } from "./PageLink";
+
+export type Props = {
+  children: ReactNode;
+  session: Session | null;
+};
+
+export const MobileShell = (props: Props) => {
+  const l10n = useL10n();
+  const [isExpanded, setIsExpanded] = useState(false);
+
+  return (
+    <div
+      className={`${styles.wrapper} ${
+        isExpanded ? styles.hasOpenMenu : styles.hasClosedMenu
+      }`}
+    >
+      <header className={styles.header}>
+        <div className={styles.headerStart}>
+          <button
+            onClick={() => setIsExpanded(!isExpanded)}
+            className={styles.menuToggleButton}
+            title={l10n.getString(
+              isExpanded
+                ? "main-nav-button-collapse-tooltip"
+                : "main-nav-button-expand-tooltip"
+            )}
+          >
+            {isExpanded ? (
+              <CloseBigIcon
+                alt={l10n.getString("main-nav-button-collapse-label")}
+              />
+            ) : (
+              <ListIcon alt={l10n.getString("main-nav-button-expand-label")} />
+            )}
+          </button>
+        </div>
+        <div className={styles.headerMiddle}>
+          <Link href="/" className={styles.homeLink}>
+            <Image
+              src={monitorLogo}
+              alt={l10n.getString("main-nav-link-home-label")}
+              width={170}
+            />
+          </Link>
+        </div>
+        <div className={styles.headerEnd}>
+          {/* For the app and user menus */}
+        </div>
+      </header>
+      <div className={styles.nonHeader}>
+        <nav className={styles.mainMenuLayer}>
+          <div className={styles.mainMenu}>
+            <ul>
+              <li>
+                <PageLink
+                  href="/redesign/user/dashboard"
+                  activeClassName={styles.isActive}
+                >
+                  {l10n.getString("main-nav-link-dashboard-label")}
+                </PageLink>
+              </li>
+              <li>
+                <a
+                  href="https://support.mozilla.org/kb/firefox-monitor-faq"
+                  title={l10n.getString("main-nav-link-faq-tooltip")}
+                >
+                  {l10n.getString("main-nav-link-faq-label")}
+                </a>
+              </li>
+            </ul>
+          </div>
+        </nav>
+        <div className={styles.content}>{props.children}</div>
```

Comment:
Not sure I understand the entire DOM layout tree&hellip; but should this be a `<div>` or `<main>`?
Or are `<main>` elements nested elsewhere in child views?

Reply:
Yeah, so two reasons it doesn't go here:

1. `.content` also includes the sidebar menu, which shouldn't be in `<main>`. (Unfortunately, I expect the sidebar and drop-down menu to be different enough that duplication is the only sane way to deal with them.)
2. Specific pages might still have non-main content inside the layout - e.g. the dashboard will also include a tab bar, the app menu, and the user menu at the top.

But good call out - the risk is that we won't mark up any content as the main content. I'm hoping to deal with this by running Axe scans on the pages during unit tests, like we do on Relay - that already caught a couple of instances where content was not in any landmark elements there.